job-tracker/src/App.js

```diff
@@ -0,0 +1,74 @@
+import React, { Component } from "react";
+import firebase from "./utils/firebase";
+
+// components
+import Header from "./layout/Header";
+import Footer from "./layout/Footer";
+import AddApplication from "./components/AddApplication";
+import EditJob from "./components/EditJob";
+import ApplicationList from "./components/ApplicationList";
+
+require("./styles/main.scss");
+
+export class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      applications: [],
+      selectedJob: {},
+      loading: false,
+      editView: false,
+    };
+  }
+
+  componentDidMount() {
+    this.setState({ loading: true });
+    const dbRef = firebase.database().ref();
+
+    dbRef.on("value", (res) => {
+      const data = res.val();
+
+      const applications = [];
+      for (let key in data) {
```

Comment:
Could probably `Array.map` over `data` here.

Reply:
res.val() returns an object so it's not possible to use an array method.